Integrations/JsonWhoIs/JsonWhoIs.py

```diff
@@ -63,34 +63,49 @@ def dict_by_ec(cur_dict: dict):
     """
     if not cur_dict:
         return None
-    return {key.capitalize(): cur_dict[key] for key in cur_dict if cur_dict[key]}
+    return {key.capitalize(): value for key, value in cur_dict.items() if cur_dict[key]}
 
 
-def list_by_ec(cur_list: list, key_needed):
+def list_by_ec(cur_list: list, needed_keys: list):
     """ Create list of dict (Json) by entry contexts convention
     Capitalize first char in dict, remove nulls, remove not needed parameters
     :param cur_list: list of dict
-    :param key_needed: key to save
+    :param needed_keys: key to save
     :return: modified list by description above
     """
     if not cur_list:
         return None
     cur_list = [createContext(index, removeNull=True) for index in cur_list]
 
     def cur_ec(index):
-        return {key.capitalize(): index[key] for key in index if key in key_needed}
+        return {key.capitalize(): index[key] for key in index if key in needed_keys}
     cur_list = [cur_ec(contact) for contact in cur_list]
     return cur_list
 
 
+def remove_none(obj):
+    """
+    Get objects and remove None or empty strings.
+    :param obj: iterable object
+    :return: Obj with None value removed
+    """
+    if isinstance(obj, (list, tuple, set)):
+        return type(obj)(remove_none(x) for x in obj if x is not (None or ''))
+    elif isinstance(obj, dict):
+        return type(obj)((remove_none(k), remove_none(v))
```

Comment:
1) how do you get a dictionary key that has none values?
2) don't use `type(obj)` do a dictionary-comprehension.

Reply:
All function removed